integration-test/src/test/java/org/apache/iotdb/confignode/it/procedure/IoTDBProcedureIT.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.iotdb.db.queryengine.plan.statement.metadata.ShowDatabaseStatement;
+import org.apache.iotdb.db.utils.constant.SqlConstant;
+import org.apache.iotdb.it.env.EnvFactory;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.apache.iotdb.confignode.procedure.impl.CreateManyDatabasesProcedure.MAX_STATE;
+import static org.apache.iotdb.consensus.ConsensusFactory.RATIS_CONSENSUS;
+
+public class IoTDBProcedureIT {
+  private static Logger LOGGER = LoggerFactory.getLogger(IoTDBProcedureIT.class);
+
+  private static final int testReplicationFactor = 2;
+
+  private static final int testConfigNodeNum = 3, testDataNodeNum = 2;
+
+  private static final long testTimePartitionInterval = 604800000;
+
+  @BeforeClass
+  public static void setUp() {
+    EnvFactory.getEnv()
+        .getConfig()
+        .getCommonConfig()
+        .setConfigNodeConsensusProtocolClass(RATIS_CONSENSUS)
+        .setSchemaRegionConsensusProtocolClass(RATIS_CONSENSUS)
+        .setDataRegionConsensusProtocolClass(RATIS_CONSENSUS)
+        //        .setSchemaReplicationFactor(testReplicationFactor)
+        .setDataReplicationFactor(1);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanClusterEnvironment();
+  }
+
+  /**
+   * During CreateManyDatabasesProcedure executing, we expect the procedure will be interrupted only
+   * once. so lets shutdown the leader at the middle of the procedure.
+   */
+  @Test
+  public void stateMachineProcedureRecoverTest() throws Exception {
+    EnvFactory.getEnv().initClusterEnvironment(3, 1);
+
+    // prepare expectedDatabases
+    Set<String> expectedDatabases = new HashSet<>();
+    for (int id = CreateManyDatabasesProcedure.getInitialStateStatic(); id < MAX_STATE; id++) {
+      expectedDatabases.add(CreateManyDatabasesProcedure.DATABASE_NAME_PREFIX + id);
+    }
+    Assert.assertEquals(MAX_STATE, expectedDatabases.size());
+
+    // prepare req
+    final TGetDatabaseReq req =
+        new TGetDatabaseReq(
+            Arrays.asList(
+                new ShowDatabaseStatement(new PartialPath(SqlConstant.getSingleRootArray()))
+                    .getPathPattern()
+                    .getNodes()),
+            SchemaConstant.ALL_MATCH_SCOPE.serialize());
+
+    SyncConfigNodeIServiceClient leaderClient =
+        (SyncConfigNodeIServiceClient) EnvFactory.getEnv().getLeaderConfigNodeConnection();
+    leaderClient.createManyDatabases();
+
+    // Check whether the procedure is ongoing, and is still far from finishing
+    TShowDatabaseResp resp = leaderClient.showDatabase(req);
+    Assert.assertTrue(0 < resp.getDatabaseInfoMap().size());
+    Assert.assertTrue(resp.getDatabaseInfoMap().size() < MAX_STATE / 2);
```

Comment:
If the databases are created too fast, will the procedure fail directly?

Reply:
This check may be redundant because IT runs in many environments. If the databases are created too fast in one or more environments, we just let them pass, and there may still be environments that are shut during procedure execution.

Reply:
How about the procedure won't finished until recovered from another ConfigNode ?

Reply:
modified, see see